core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -271,6 +271,9 @@ public boolean requiresAnalysisSettings() {
             Settings currentSettings = entry.getValue();
             String typeName = currentSettings.get("type");
             if (analyzer) {
+                if(defaultInstance.containsKey(name)) {
+                    throw new IllegalArgumentException("do not allow custom analyzers to have the same names [" + name + "] as built-in analyzers");
+                }
```

Comment:
We need to only enforce this on new indices, can you add a check on the index version? (available in the index settings)